### DISCUSSION TITLE
How to prevent trailing slash on baseURL in Nuxt 3?

### DISCUSSION THREAD

**Discussion by FilipBartos** [nuxt/nuxt#29229] [Questions]
Hello everyone 🙋🏻‍♂️,

I'm working on a clean Nuxt 3 project with a custom baseURL configured as follows:

```js
app: {
  baseURL: '/custom-base-url',
}
```

I have two pages, `index.vue` and `about.vue`, in the pages directory. When I launch the project and visit `hostname/custom-base-url`, a trailing slash is automatically added, changing the URL to `hostname/custom-base-url/`. However, the about page works as expected and has the URL `hostname/custom-base-url/about`.

Is there a way to prevent the trailing slash from being added to the base URL so that the URL remains `hostname/custom-base-url` without the / at the end?

Interestingly, when I remove the pages directory and create an `app.vue` file, the trailing slash is no longer added, and the URL stays as `hostname/custom-base-url`.

It can be reproduced at https://stackblitz.com/edit/nuxt-starter-q86cfv?file=nuxt.config.ts.

Any advice would be appreciated 🙏

Thanks!

**Comment by oneminch** [comment]
I don't think it's possible. The `baseURL` configuration automatically adds a trailing slash by default, and it can't be easily overridden using configuration. This is the closest config option I found from the Vue Router docs ([`router.options.strict`](https://router.vuejs.org/api/interfaces/RouterOptions.html#strict)), but I ran a simple Nuxt project to test it out, and it didn't work.

Have you tried defining a route middleware using `defineNuxtRouteMiddleware`?
